examples/inspection/plot_partial_dependence.py

```diff
@@ -567,3 +574,48 @@
 plt.show()
 
 # %%
+# Interaction strength
+# --------------------
+#
+# Above considerations show that comparing 2D PDPs with their univariate versions
+# give hints about feature interactions. This idea is formalized by Friedman
+# and Popescu's (pairwise) H-statistic, see [4]_.
+# It measures how well the 2D partial dependence function can be approximated
+# by the two one-dimensional partial dependence functions.
+# The resulting value is then normalized and can be interpreted as proportion of effect
+# variability explained by the interaction. Besides this relative measure, we advocate
+# to also consider unnormalized statistics. They can directly be compared between
+# feature pairs to see which interactions are strongest.
+#
+# Since computational burden is high, H-statistics are usually calculated only for
+# important features, e.g., selected by permutation importance. What do we get for
+# our model?
+print("Select five important features and crunch H-statistics...")
+
+from sklearn.inspection import h_statistic, permutation_importance
+
+tic = time()
+imp = permutation_importance(hgbdt_model, X=X_train, y=y_train, random_state=0)
+features = X_train.columns[np.argsort(imp.importances_mean)][-5:]
+
+H = h_statistic(hgbdt_model, X=X_train, features=features, random_state=0)
+
+print(f"done in {time() - tic:.3f}s")
+
+fig, axes = plt.subplots(1, 2, figsize=(8, 4))
+bar_labels = np.array([str(pair) for pair in H["feature_pairs"]])
```

Comment:
```suggestion
bar_labels = [str(pair) for pair in H["feature_pairs"]]
```
No need for an ndarray here.

Reply:
We need to sort the labels later in the same order as the values. So I am undoing this change. (Actually, a corresponding pandas code would be extremely compact.)

Reply:
I see, so an array makes sense.

A lot of things would be easier by choosing a single df library, but we want to stay df agnostic or work for several df libraries, like pandas and polars.